dependencies.yaml

```
@@ -287,6 +287,7 @@ dependencies:
       - output_types: conda
         packages:
           - fmt>=10.1.1,<11
+          - flatbuffers
```

Add a pinning that matches the CMake pinning: https://github.com/rapidsai/cudf/blob/79a1eed785fccbca2c20ff5cc844ec1a9e741ee5/cpp/cmake/thirdparty/get_flatbuffers.cmake#L33

I don't understand how the [flatbuffers versioning philosophy](https://github.com/google/flatbuffers/wiki/Versioning) impacts compatibility, so we better pin to exactly the same version.
```suggestion
          - flatbuffers==24.3.25
```

---

We actually need that particular version due to a static assert in the generated code here
https://github.com/rapidsai/cudf/blob/branch-24.10/cpp/src/io/parquet/ipc/Message_generated.h#L10